compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1467,6 +1467,11 @@ fn add_local_crate_metadata_objects(
         {
             cmd.add_object(obj);
         }
+    } else if crate_type == CrateType::Cdylib {
+        if let Some(obj) = codegen_results.metadata_module.as_ref().and_then(|m| m.object.as_ref())
+        {
+            cmd.add_object(obj);
```

Comment:
A cdylib must not contain the crate metadata. It is the whole point of cdylibs to be a regular dylib as produced by a C compiler, not a dylib that can be used like regular rust crates.

Reply:
I figured I'd enable it and then excluded all but that one svh... but maybe instead I should just try and replicate what the metadatamodule is doing... (btw, going to rebase everything, so I've got cranelift in there.)

Reply:
Ok so it looks like rust crates can't depend on cdylibs so this is now out of scope. (Have rolled back the change)